pydis_core/_bot.py

```diff
@@ -231,6 +231,12 @@ async def wait_until_guild_available(self) -> None:
         """
         await self._guild_available.wait()
 
+    async def process_commands(self, message: discord.Message) -> None:
+        """Wait until all extensions are loaded before processing commands."""
+        if self._extension_loading_task:
```

Comment:
Wouldn't it be better to put the condition check and the awaiting in `wait_until_extensions_loaded ` like you had before, and reuse it in the 2 callsites ?

Reply:
If you're referring to `_sync_app_commands` as the other call site, I don't think that's relevant, since that function is only called after creating the task, so there's no point in check the task exists.